virtcontainers/network.go

```diff
@@ -417,7 +417,29 @@ func (endpoint *PhysicalEndpoint) Detach(netNsCreated bool, netNsPath string) er
 
 // HotAttach for physical endpoint not supported yet
 func (endpoint *PhysicalEndpoint) HotAttach(h hypervisor) error {
-	return fmt.Errorf("PhysicalEndpoint does not support Hot attach")
+	networkLogger().WithField("endpoint-type", "physical").Info("Attaching endpoint (hotplug)")
+
+	// Unbind physical interface from host driver and bind to vfio
+	// so that it can be passed to qemu.
+	if err := bindNICToVFIO(endpoint); err != nil {
+		return err
+	}
+
+	// Generate a unique ID to be used for hypervisor commandline fields
+	randBytes, err := utils.GenerateRandomBytes(8)
+	if err != nil {
+		return err
+	}
+	id := hex.EncodeToString(randBytes)
+
+	// TODO: use device manager as general device management entrance
+	d := config.VFIODev{
+		ID:  id,
+		BDF: endpoint.BDF,
+	}
+
+	_, err = h.hotplugAddDevice(d, vfioDev)
```

Comment:
Since you are doing the rework, how about we use `sandbox.DeviceManager` to manage vfio devices?

e.g.
```
dev, err := sandbox.DeviceManager.NewDevice(deviceInfo{})

sandbox.DeviceManager.AttachDevice(dev.DeviceID(), sandbox)
```

This can gurantee we manage devices with same struct and after container exits, every devices can be detached correctly.

This will also need some rework as PhysicalEndpoint doesn't have a `sandbox` field

Reply:
@WeiZhang555 I wanted to enable VFIO net hotplug support in order to support #600 but now that I have posted #623 , I think I can spend more time to actually do this PR more properly.
I'll try to spend some time to use the DeviceManager, so that I can implement the detach too ;)